promql/value.go

```diff
@@ -264,10 +296,11 @@ func (m Matrix) String() string {
 }
 
 // TotalSamples returns the total number of samples in the series within a matrix.
+// It takes into account the number of samples in the histograms using the histogramSize method.
```

Comment:
I would just repeat the same sentence as you have done above for `func (vec Vector) TotalSamples()`, i.e. 
```
// Float samples have a weight of 1 in this number, while histogram samples have a higher
// weight according to their size compared with the size of a float sample.
// See HPoint.histogramSize for details.
```